add never_triggered

Mariusthvdb opened the pull request:
Add never_triggered to the last-triggered secondary_info.
as it stands, the secondary_info line is empty when used on an automation or script that hasn't been triggered yet, while the user expects info on that line.

my suggestion would be to add:
```
"automation": {
        "last_triggered": "Last triggered",
        "never_triggered":"Never triggered",
        "trigger": "Execute"
      }
```
and
```
"script": {
        "last_triggered": "Last triggered",
        "never_triggered":"Never triggered",
        "execute": "Execute"
      }
```
to the [translation file](https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/translations/en.json).

Mariusthvdb committed:
add never_triggered

Add never_triggered to the last-triggered secondary_info.
as it stands, the secondary_info line is empty when used on an automation or script that hasn't been triggered yet, while the user expects info on that line.

bramkragten commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,15 @@ class HuiGenericEntityRow extends LitElement {
+                     .datetime=${stateObj.attributes.last_triggered}
+                   ></ha-relative-time>
+                 `
+               : this.hass.localize("ui.panel.lovelace.entities.never_triggered")
```
You should add this to https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/translations/en.json

Mariusthvdb replied:
ok I will, referring to this PR.

bramkragten replied:
No, in this PR

Mariusthvdb replied:
see https://github.com/home-assistant/home-assistant-polymer/pull/4444

iantrich replied:
It should be in this PR, not a separate one

Mariusthvdb replied:
ok, I will, if I can find how to do that... how can I add a second file to this PR to edit?

bramkragten replied:
Just edit it in the same branch: https://github.com/Mariusthvdb/home-assistant-polymer/tree/patch-3

You would also need to fix the lint issues (running prettier would fix that for you)

Mariusthvdb reviewed (commented):
thanks for your feedback, appreciated!

Mariusthvdb committed:
add "never_triggered"

for automation and script, to use in secondary_info
add last_triggered on script, because it was missing

iantrich commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,15 @@ class HuiGenericEntityRow extends LitElement {
+                     .datetime=${stateObj.attributes.last_triggered}
+                   ></ha-relative-time>
+                 `
+               : this.hass.localize("ui.panel.lovelace.entities.never_triggered")
```
this path doesn't match what you added to en.json

Mariusthvdb replied:
ok so now I am confused. Please bare with me..
I believed you suggested this path, and so I added that to the pr of the hui-generic-entity-row.ts

following that Bram asked me to add to the translation file https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/translations/en.json

which of both should I change?

bramkragten replied:
It is the right file it is just that `ui.panel.lovelace.entities.never_triggered` isn't what you created that is: `ui.card.automation.never_triggered`

bramkragten replied:
And you should not add keys that we don't use in the code, like the other keys you added

Mariusthvdb replied:
I see, I searched for automation and script, but should have looked for path lovelace is it? I think this is the only hat to lovelace I can find, is that what you refer to? https://github.com/home-assistant/home-assistant-polymer/blob/4e5406b27b0a14788e80859a96d2b54871954b39/src/translations/en.json#L1575 ?

if so, there is no path to entities yet? I should add this:
"entitities": {"never_triggered":"Never triggered"},

bramkragten replied:
Put it in here:
https://github.com/home-assistant/home-assistant-polymer/blob/4e5406b27b0a14788e80859a96d2b54871954b39/src/translations/en.json#L1700

Mariusthvdb committed:
delete spaces

Mariusthvdb committed:
add 2 spaces

travis check

Mariusthvdb committed:
travis check row 116

iantrich commented on src/translations/en.json:
```diff
@@ -458,6 +459,8 @@
       },
       "script": {
+        "last_triggered": "Last triggered",
+        "never_triggered":"Never triggered",
```
Remove these two. You are not using them.

iantrich commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,17 @@ class HuiGenericEntityRow extends LitElement {
+                    ></ha-relative-time>
+                  `
+                : this.hass.localize(
+                    "ui.panel.lovelace.entities.never_triggered"
```
The path you pass is still not the same as what you out in en.json

Mariusthvdb replied:
Ive just corrected that, deleted the incorrect additions and added never_triggered to line 1700 in en.json, following Brams instruction

Mariusthvdb committed:
add never_triggered

to entities, delete incorrect entries to automation and script

bramkragten commented on src/translations/en.json:
```diff
@@ -1698,6 +1698,7 @@
               "optional": "Optional"
             },
             "entities": {
+              "never_triggered":"Never triggered",
```
```suggestion
              "never_triggered": "Never triggered",
```

Mariusthvdb committed:
Update src/translations/en.json

Co-Authored-By: Bram Kragten <mail@bramkragten.nl>

bramkragten commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,17 @@ class HuiGenericEntityRow extends LitElement {
+                    ></ha-relative-time>
+                  `
+                : this.hass.localize(
+                    "ui.panel.lovelace.entities.never_triggered"
```
```suggestion
                    "ui.panel.lovelace.cards.entities.never_triggered"
```

Mariusthvdb committed:
Update src/panels/lovelace/components/hui-generic-entity-row.ts

Co-Authored-By: Bram Kragten <mail@bramkragten.nl>

bramkragten commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,17 @@ class HuiGenericEntityRow extends LitElement {
+                    ></ha-relative-time>
+                  `
+                : this.hass.localize(
+                    "ui.panel.lovelace.cards.entities.never_triggered"
```
```suggestion
                    "ui.panel.lovelace.card.entities.never_triggered"
```

bramkragten commented:
Please also test your PR

Mariusthvdb committed:
Update src/panels/lovelace/components/hui-generic-entity-row.ts

Co-Authored-By: Bram Kragten <mail@bramkragten.nl>

Mariusthvdb commented:
> Please also test your PR

ok let me know how?

bramkragten commented:
Follow there instructions: https://developers.home-assistant.io/docs/en/frontend_development.html

Mariusthvdb commented:
That is a bit overtaxing at the moment I fear... setting up a full development environment I wouldn't know where to start with, for such a small edit..
Hope to have a spare device for that anytime soon, please have patience.

ludeeus commented on src/panels/lovelace/components/hui-generic-entity-row.ts:
```diff
@@ -105,14 +105,17 @@ class HuiGenericEntityRow extends LitElement {
+                : this.hass.localize(
+                    "ui.panel.lovelace.card.entities.never_triggered"
+                  )
               : ""}
```
```suggestion
              }
```

Mariusthvdb replied:
Hi Ludeeus,

This was already in the original file. Should it be taken out ?

ludeeus replied:
No, just looked a little off with all the nesting.

Mariusthvdb commented:
@bramkragten 
having some difficulties setting up my dev environment, which won't go beyond setting up an integration (Sonos) and doesn't finish the full startup sequence... because of this, I can't set up any testing yet I fear. Is there any way you guys can test if this works now, so we can proceed with the PR?

ludeeus commented:
Tested, it does not work, probably needs a check against `null` and/or `undefined`

Mariusthvdb commented:
a check is what I had in my first suggestion, which was later on superseded with Ian 's code. 

what would be best? to first check if not null? like
```
              : this.config.secondary_info === "last-triggered" &&
                 stateObj.attributes.last_triggered !== null
                 ? html`
                     <ha-relative-time
                       .hass=${this.hass}
                       .datetime=${stateObj.attributes.last_triggered}
                     ></ha-relative-time>
                   `
                 : this.hass.localize(
                     "ui.panel.lovelace.card.entities.never_triggered"
                   )
```

bramkragten commented:
Probably the second with a falsy check instead of `null` (as you now don't catch undefined)

Mariusthvdb commented:
since this is new to me, I'm not sure what I should add. Checked all never_triggered automations and scripts in HA states, and in template editor. 

In states they all `show last_triggered: null`, while in templates attributes_last_triggered == None or none. Never unknown or null.

<img width="553" alt="Schermafbeelding 2020-01-13 om 13 40 49" src="https://user-images.githubusercontent.com/33354141/72256968-c3c10080-360a-11ea-8573-f324eba92659.png">

<img width="1160" alt="Schermafbeelding 2020-01-13 om 13 42 32" src="https://user-images.githubusercontent.com/33354141/72256980-c91e4b00-360a-11ea-94a0-b552b6c4cfa2.png">

If that still leaves you needing an extra falsy check, please let me know what that would need to be besides testing for `!== null`

iantrich commented:
The problem is not the check it is your translation path. You added your translation to ui.panel.lovelace.editor.entities.never_triggered in `en.json`

iantrich committed:
Update en.json

iantrich committed:
Update hui-generic-entity-row.ts

Mariusthvdb commented:
wow, thanks Ian, really appreciated!
was seriously confused by all of this, yet so simple....
glad it is finally approved!
